File: Checks/PWD002/example-omp.c
// PWD002: Unprotected multithreading reduction operation

void example(int *array) {
  int sum = 0;

  #pragma omp parallel for default(none) shared(array, sum)
  for (int i = 0; i < 10; i++) {
    sum += array[i];
  }
}

File: Checks/PWD002/example-omp.f90
! PWD002: Unprotected multithreading reduction operation

subroutine example(array)
  integer, intent(in) :: array(:)
  integer :: i, sum

  sum = 0

  !$omp parallel do default(none) shared(array, sum)
  do i = 1, size(array, 1)
    sum = sum + array(i)
  end do
end subroutine example
